src/mastercard-auth.js

```diff
@@ -16,8 +16,23 @@ module.exports = function (context) {
 
   const mastercard = context.request.getEnvironmentVariable('mastercard');
 
-  if (mastercard && commaDecodedUrl.includes('mastercard.com')) {
+  if(mastercard){
```

Comment:
If possible, I would return early instead of having a `if` with 60 lines of code in it:

```javascript
if (!mastercard) {
    // No mastercard configuration in the current environment, do nothing
    return
}
```